bin/templates/cordova/Api.js

```diff
@@ -212,6 +212,11 @@ Api.prototype.addPlugin = function (plugin, installOptions) {
       installOptions.android_studio = true;
     }
 
+    //CB-11964: Do a clean when installing the plugin code to get around
+    //the Gradle bug introduced by the Android Gradle Plugin Version 2.2
+    //TODO: Delete when the next version of Android Gradle plugin comes out
+    
+    this.clean();
```

Comment:
Is it out of promise chain intentionally? I think that `this.clean()` returns a promise.

I've tried to merge it to the promise chain, but got the following exception upon `cordova platform add`:
```
Error: ENOENT: no such file or directory, open 'C:\Cordova\foo\platforms\android\assets\www\cordova_plugins.js'
    at Error (native)
    at Object.fs.openSync (fs.js:549:18)
    at Object.fs.writeFileSync (fs.js:1156:15)
    at C:\Cordova\foo\platforms\android\cordova\node_modules\cordova-common\src\PluginManager.js:137:12
    at _fulfilled (C:\Cordova\foo\platforms\android\cordova\node_modules\q\q.js:834:54)
    at self.promiseDispatch.done (C:\Cordova\foo\platforms\android\cordova\node_modules\q\q.js:863:30)
    at Promise.promise.promiseDispatch (C:\Cordova\foo\platforms\android\cordova\node_modules\q\q.js:796:13)
    at C:\Cordova\foo\platforms\android\cordova\node_modules\q\q.js:857:14
    at runSingle (C:\Cordova\foo\platforms\android\cordova\node_modules\q\q.js:137:13)
    at flush (C:\Cordova\foo\platforms\android\cordova\node_modules\q\q.js:125:13)
```

I wonder why it works the way you put it and if it is dangerous to leave it like this.